__tests__/app.test.js

```diff
@@ -518,4 +518,152 @@ describe("GET /api/users test suite", () => {
+      const articlesArray = data.body.articles;
+      expect(articlesArray).toBeSortedBy("author", { descending: true });
+    });
+  });
+  
+  test("when we get 'sort_by=title&order=ASC', returned data is sorted by title ascending", () => {
+    return request(app)
+    .get("/api/articles?sort_by=title&order=ASC")
+    .expect(200)
+    .then((data) => {
+      const articlesArray = data.body.articles;
+      expect(articlesArray).toBeSortedBy("title", { ascending: true });
+    });
+  });
+  
+  test("when we get 'order=ASC&sort_by=votes', returned data is sorted by votes ascending", () => {
+    return request(app)
+    .get("/api/articles?order=ASC&sort_by=votes")
+    .expect(200)
+    .then((data) => {
+      const articlesArray = data.body.articles;
+      expect(articlesArray).toBeSortedBy("votes", { ascending: true });
+    });
+  });
+
+  test("if sort_by=BAD_REQUEST then a 400 Bad request is received", () => {
+    return request(app)
+    .get("/api/articles?sort_by=BAD_REQUEST77")
+    .expect(400)
+    .then((data) => {
+      expect(data.body).toEqual({ msg: "Bad Request - Invalid query sort_by= was provided" });
+    });
+  });
+});
+
+// NOTES on test data:
+// 'mitch' or 'cats' or 'paper' are the 3 topics in the topics db
+// in articles the hits for those are mitch(11), cats(1), paper(0)
+describe("GET with TOPIC=: /api/articles?topic=", () => {
+  test("when topic is omitted - returns all 12 articles in the db", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((data) => {
+        expect(data.body.articles.length).toEqual(12);
+      });
+  });
+
+  test("topic = mitch returns 11 articles", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then((data) => {
+        expect(data.body.articles.length).toEqual(11);
+      });  
+    });    
+    
+    test("topic = paper returns zero articles - but is still a 200 status code and not an error", () => {
+      return request(app)
+      .get("/api/articles?topic=paper")
+      .expect(200)
+      .then((data) => {
+        expect(data.body.articles.length).toEqual(0);
+      });  
+    });    
+    
+    test("all 3 queries work together: ?topic=mitch&order=ASC&sort_by=votes  ====>  11 articles sorted by votes in ascending order", () => {
+      return request(app)
+      .get("/api/articles?topic=mitch&order=ASC&sort_by=votes")
+      .expect(200)
+      .then((data) => {
+        const articlesArray = data.body.articles;
+        expect(articlesArray.length).toEqual(11);
+        expect(articlesArray).toBeSortedBy("votes", { ascending: true });
+      });  
+    });    
+    
+    test("when topic is not in the db, get an empty array and 200 status code", () => {
+      return request(app)
+      .get("/api/articles?topic=A_TOPIC_NOT_IN_DB")
+      .expect(200)
+      .then((data) => {
+        expect(data.body.articles.length).toEqual(0);
+      });
+  });
+});
+
```

Comment:
Could we just add one more test here for a 404? If there's a non-existent topic query, e.g. topic=bananas